packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -263,13 +269,15 @@ export async function accumulateMetadata(
 
     parentPromise = parentPromise.then((resolved) => {
       return layerMetadataPromise.then((metadata) => {
+        resolved = deepClone(resolved)
```

Comment:
This adds a lot of copying in dev. We can always merge into `resolvedMetadata` instead of what was resolved in the parentPromise so we never try to merge into a frozen object. the clone/freeze should only affect the value passed to generateMetadata but we want the code path for actual merging to be identical in dev/prod so we have higher confidence it is working when we test it in dev. There is always a risk when we do env specific code that a logic change means divergent behavior. We should limit that as much as possible to avoid pain in the future